lib/tasks/migrations/migrations.rake

```diff
@@ -74,6 +74,44 @@ namespace :migrations do
     exit has_encountred_issue
   end
 
+  task :rooms, [:start, :stop] => :environment do |_task, args|
+    start, stop = range(args)
+    has_encountred_issue = 0
+
+    Room.select(:uid, :name, :bbb_id, :last_session, :user_id)
+        .find_each(start: start, finish: stop, batch_size: COMMON[:batch_size]).each do |r|
+          params = { room: { friendly_id: r.uid,
+                             name: r.name,
+                             meeting_id: r.bbb_id,
+                             last_session: r.last_session.to_datetime,
```

Comment:
This will cause an issue for rooms that has no `last_session`,
Besides, the conversion to `DateTime` isn't required we can marshal the `ActiveSupport::TimeWithZone` object directly and in both cases time zome will be conserved,
However, this is more then 2 times faster then if we marshal the `ActiveSupport::TimeWithZone` so good job on that!

Reply:
Umm, not sure what you mean.
As I recall both works, the `last_session` will be represented as an `ActiveSupport::TimeWithZone` on both versions.
However, converting that to a `DateTime` instance seems to be better for marshalling.
You can leave it as is and better handle the `Nil` case or sacrifice a bit of performance by omitting the conversion.
You can further test the task on your local setup,

Reply:
Added a safe operator on last_session.
I will run a test by the end of the week.

Reply:
No worries, I'm already testing it.